packages/core/contracts/common/implementation/FixedPoint.sol

```diff
@@ -373,4 +377,386 @@ library FixedPoint {
+     */
+    function sub(Signed memory a, Signed memory b) internal pure returns (Signed memory) {
+        return Signed(a.rawValue.sub(b.rawValue));
+    }
+
+    /**
+     * @notice Subtracts an unscaled int256 from an `Signed`, reverting on overflow.
+     * @param a a FixedPoint.Signed.
+     * @param b an int256.
+     * @return the difference of `a` and `b`.
+     */
+    function sub(Signed memory a, int256 b) internal pure returns (Signed memory) {
+        return sub(a, fromUnscaledInt(b));
+    }
+
+    /**
+     * @notice Subtracts an `Signed` from an unscaled int256, reverting on overflow.
+     * @param a an int256.
+     * @param b a FixedPoint.Signed.
+     * @return the difference of `a` and `b`.
+     */
+    function sub(int256 a, Signed memory b) internal pure returns (Signed memory) {
+        return sub(fromUnscaledInt(a), b);
+    }
+
+    /**
+     * @notice Multiplies two `Signed`s, reverting on overflow.
+     * @dev This will "floor" the product.
+     * @param a a FixedPoint.Signed.
+     * @param b a FixedPoint.Signed.
+     * @return the product of `a` and `b`.
+     */
+    function mul(Signed memory a, Signed memory b) internal pure returns (Signed memory) {
+        // There are two caveats with this computation:
+        // 1. Max output for the represented number is ~10^41, otherwise an intermediate value overflows. 10^41 is
+        // stored internally as an int256 ~10^59.
+        // 2. Results that can't be represented exactly are truncated not rounded. E.g., 1.4 * 2e-18 = 2.8e-18, which
+        // would round to 3, but this computation produces the result 2.
+        // No need to use SafeMath because SFP_SCALING_FACTOR != 0.
+        return Signed(a.rawValue.mul(b.rawValue) / SFP_SCALING_FACTOR);
+    }
+
+    /**
+     * @notice Multiplies an `Signed` and an unscaled int256, reverting on overflow.
+     * @dev This will "floor" the product.
+     * @param a a FixedPoint.Signed.
+     * @param b an int256.
+     * @return the product of `a` and `b`.
+     */
+    function mul(Signed memory a, int256 b) internal pure returns (Signed memory) {
+        return Signed(a.rawValue.mul(b));
+    }
+
+    /**
+     * @notice Multiplies two `Signed`s and "ceil's" the product, reverting on overflow.
+     * @param a a FixedPoint.Signed.
+     * @param b a FixedPoint.Signed.
+     * @return the product of `a` and `b`.
+     */
+    function mulAwayFromZero(Signed memory a, Signed memory b) internal pure returns (Signed memory) {
+        int256 mulRaw = a.rawValue.mul(b.rawValue);
+        int256 mulTowardsZero = mulRaw / SFP_SCALING_FACTOR;
+        // Manual mod because SignedSafeMath doesn't support it.
+        int256 mod = mulRaw % SFP_SCALING_FACTOR;
+        if (mod != 0) {
+            bool isResultPositive = isLessThan(a, 0) == isLessThan(b, 0);
+            int256 valueToAdd = isResultPositive ? int256(1) : int256(-1);
+            return Signed(mulTowardsZero.add(valueToAdd));
```

Comment:
Note: this extra complexity is necessary to handle the case where the division rounds to 0. In that case the signs of the operands needs to be considered.